vite-project/src/Form.tsx

```diff
@@ -0,0 +1,254 @@
+import React from "react";
+import "./Form.css";
+
+interface FormProps {
+  onSubmit: (data: FormData) => void;
+}
+
+const initialFormData = {
+  brand: "",
+  model: "",
+  year: 0,
+  body_type: "",
+  mileage_km: 0,
+  gearbox: "",
+  fuel: "",
+  price_per_day: 0,
+  horse_power: 0,
+  engine_capacity: 0,
+  purpose: "",
+};
+
+interface FormData {
+  brand: string;
+  model: string;
+  year: number | null;
+  body_type: string;
+  mileage_km: number;
+  gearbox: string;
+  fuel: string;
+  price_per_day: number;
+  horse_power: number;
+  engine_capacity: number;
+  purpose: string;
+}
+
+function Form({ onSubmit }: FormProps) {
+  const [formData, setFormData] = React.useState<FormData>(initialFormData);
+
+  const [errors, setErrors] = React.useState<Partial<FormData>>({});
+
+  function handleInputChange(event: React.ChangeEvent<HTMLInputElement>) {
+    const { name, value } = event.target;
+    setFormData({ ...formData, [name]: value });
+    setErrors({ ...errors, [name]: "" });
+  }
+
+  function validateForm() {
+    const newErrors: Partial<FormData> = {};
+
+    if (!formData.brand.trim()) {
+      newErrors.brand = "Brand is required";
+    }
+
+    setErrors(newErrors);
+    return Object.keys(newErrors).length === 0;
+  }
+
+  const handleReset = () => {
+    setFormData(initialFormData);
+  };
+
+  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    if (validateForm()) {
+      onSubmit(formData);
+    }
+  }
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <h2>Car adding form</h2>
+      <label>
+        Brand:
+        <input
+          type="text"
+          name="brand"
+          value={formData.brand}
+          onChange={handleInputChange}
+          required
+        />
+      </label>
+      <label>
+        Model:
+        <input
+          type="text"
+          name="model"
+          value={formData.model}
+          onChange={handleInputChange}
+          required
+        />
+      </label>
+
+      <label>
+        Year:
+        <input
+          type="number"
+          name="year"
+          value={formData.year}
+          onChange={handleInputChange}
+          min="2000"
+          max="2024"
+          required
+        />
+      </label>
+
+      <label>
+        Body Type:
+        <select
+          name="body_type"
+          value={formData.body_type}
+          onChange={handleInputChange}
+          required
+        >
+          <option value="">Select...</option>
+          <option value="Sedan">Sedan</option>
+          <option value="SUV">SUV</option>
+          <option value="Hatchback">Hatchback</option>
+          <option value="Convertible">Convertible</option>
+          <option value="Coupe">Coupe</option>
```

Comment:
What do you think about creating an array, for example `bodyTypeOptions`, and use array `map` function for rendering it? In this case, it will be easier for you to add new options for your select or update an existing one if necessary - everything will be in one array and will be automatically displayed on the page